### PR TITLE
Issue #92 - Refactor Flow and Job to StateMachine and Execution

### PR DESCRIPTION
**Pull request by mtcotton**

### CODE REVIEW

**Comment by mtcotton** [comment]
Saving as draft until I can test with a clean project, but junit and mlunittests pass

File: state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/custom-steps-state-machine.asl.json [hunk 1]
```diff
@@ -10,7 +10,7 @@
       "Comment": "runs the dhf 5 flow step",
       "Resource": "/state-conductor/actions/common/dhf/dhf5RunFlowStepAction.sjs",
       "Parameters": {
-        "flowName": "CustomFlow",
+        "name": "CustomFlow",
```
Comment (aclavio) [inline state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/custom-steps-state-machine.asl.json:13]: rollback to keep DHF naming convention of "flowName"

File: state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/missing-dhf-state-machine.asl.json [hunk 2]
```diff
@@ -10,7 +10,7 @@
       "Comment": "runs the dhf 5 flow step",
       "Resource": "/state-conductor/actions/common/dhf/dhf5RunFlowStepAction.sjs",
       "Parameters": {
-        "flowName": "MissingFlow",
+        "name": "MissingFlow",
```
Comment (aclavio) [inline state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/missing-dhf-state-machine.asl.json:13]: rollback to keep DHF naming convention of "flowName"

File: state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-state-machine.asl.json [hunk 3]
```diff
@@ -10,7 +10,7 @@
       "Comment": "runs the dhf 5 flow",
       "Resource": "/state-conductor/actions/common/dhf/dhf5RunFlowAction.sjs",
       "Parameters": {
-        "flowName": "PersonFlow",
+        "name": "PersonFlow",
```
Comment (aclavio) [inline state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-state-machine.asl.json:13]: rollback to keep DHF naming convention of "flowName"

File: state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-steps-state-machine.asl.json [hunk 4]
```diff
@@ -10,7 +10,7 @@
       "Comment": "runs the dhf 5 flow step",
       "Resource": "/state-conductor/actions/common/dhf/dhf5RunFlowStepAction.sjs",
       "Parameters": {
-        "flowName": "PersonFlow",
+        "name": "PersonFlow",
```
Comment (aclavio) [inline state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-steps-state-machine.asl.json:13]: rollback to keep DHF naming convention of "flowName"

File: state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-steps-state-machine.asl.json [hunk 5]
```diff
@@ -27,7 +27,7 @@
...
```
Comment (aclavio) [inline state-conductor-dhf5-example/src/test/ml-modules/root/test/suites/StateConductorDHF5Suite/test-data/stateMachines/person-steps-state-machine.asl.json:30]: rollback to keep DHF naming convention of "flowName"

**Review by aclavio — CHANGES_REQUESTED** [review]

**Pull request merged**